### PR TITLE
Wi-Fi: Fix scan

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c [hunk 1]
```diff
@@ -452,6 +452,16 @@ int wifi_nrf_wpa_supp_scan2(void *if_priv, struct wpa_driver_scan_params *params
 	scan_info.scan_mode = 0;
 	scan_info.scan_reason = SCAN_CONNECT;
 
+	/* Copy extra_ies */
+	if (params->extra_ies_len && params->extra_ies_len <= IMG_MAX_IE_LEN) {
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:456]: Change IMG_MAX_IE_LEN to NRF_WIFI_MAX_IE_LEN

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c [hunk 2]
```diff
@@ -452,6 +452,16 @@ int wifi_nrf_wpa_supp_scan2(void *if_priv, struct wpa_driver_scan_params *params
...
+		memcpy(scan_info.scan_params.ie.ie, params->extra_ies, params->extra_ies_len);
+		scan_info.scan_params.ie.ie_len = params->extra_ies_len;
+	} else if (params->extra_ies_len) {
+		LOG_ERR("%s: extra_ies_len %d is greater than max IE len %d\n",
+			__func__, params->extra_ies_len, IMG_MAX_IE_LEN);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:461]: Change IMG_MAX_IE_LEN to NRF_WIFI_MAX_IE_LEN

File: drivers/wifi/nrf700x/zephyr/src/zephyr_disp_scan.c [hunk 3]
```diff
@@ -47,6 +47,10 @@ int wifi_nrf_disp_scan_zep(const struct device *dev,
 
 	scan_info.scan_mode = AUTO_SCAN;
 	scan_info.scan_reason = SCAN_DISPLAY;
+	/* Wildcard SSID to trigger active scan */
+	scan_info.scan_params.num_scan_ssids = 1;
+	scan_info.scan_params.scan_ssids[0].img_ssid_len = 0;
+	scan_info.scan_params.scan_ssids[0].img_ssid[0] = 0;
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_disp_scan.c:54]: Remove img_ prefixes

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c (hunk 1 above)
Comment (krish2718) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:456]: These are not introduced in this PR, and also present in common header files with RPU, so, left it untouched, once we clean those header files then we can do these changes.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c (hunk 2 above)
Comment (krish2718) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:461]: These are not introduced in this PR, and also present in common header files with RPU, so, left it untouched, once we clean those header files then we can do these changes.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_disp_scan.c (hunk 3 above)
Comment (krish2718) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_disp_scan.c:54]: These are not introduced in this PR, and also present in common header files with RPU, so, left it untouched, once we clean those header files then we can do these changes.

**Review by sachinthegreen — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**